src/bin/stratisd.rs

```diff
@@ -102,47 +100,37 @@ fn initialize_log(debug: bool) -> buff_log::Handle<env_logger::Logger> {
     }
 }
 
-/// To ensure only one instance of stratisd runs at a time, acquire an
-/// exclusive lock. Return an error if lock attempt fails.
-fn trylock_pid_file() -> StratisResult<File> {
-    let mut f = match OpenOptions::new()
-        .read(true)
-        .write(true)
-        .create(true)
-        .open(STRATISD_PID_PATH)
-    {
-        Ok(f) => f,
-        Err(e) => {
-            if e.kind() == ErrorKind::PermissionDenied {
-                return Err(StratisError::Error(
-                    "Must be running as root in order to start daemon.".to_string(),
-                ));
-            }
-            return Err(e.into());
-        }
-    };
-    match flock(f.as_raw_fd(), FlockArg::LockExclusiveNonblock) {
-        Ok(_) => {
-            f.write_all(format!("{}\n", getpid()).as_bytes())?;
-            Ok(f)
-        }
-        Err(_) => {
-            let mut buf = String::new();
-            f.read_to_string(&mut buf)?;
-            // pidfile is supposed to contain pid of holder. But you never
-            // know so be paranoid.
-            let pid_str = buf
-                .split_whitespace()
-                .next()
-                .and_then(|s| s.parse::<pid_t>().ok())
-                .map(|pid| format!("{}", pid))
-                .unwrap_or_else(|| "<unknown>".into());
-            Err(StratisError::Error(format!(
-                "Daemon already running with pid: {}",
-                pid_str
-            )))
-        }
+/// To ensure only one instance of stratisd runs at a time, create a socket
+/// with a specified abstract namespace path. Return an error if socket creation
+/// attempt fails.
+fn try_abstract_socket() -> StratisResult<RawFd> {
+    if !Uid::effective().is_root() {
+        return Err(StratisError::Error(
+            "Must be running as root in order to start daemon.".to_string(),
+        ));
     }
+    let sock_fd = socket(
+        AddressFamily::Unix,
+        SockType::Stream,
+        SockFlag::empty(),
+        None,
+    )
+    .map_err(|e| {
+        StratisError::Error(format!(
+            "Failed to create socket and test if process is already running: {}",
+            e
+        ))
+    })?;
+    bind(sock_fd, &SockAddr::Unix(match UnixAddr::new_abstract(STRATISD_SOCKET_ADDR) {
+        Ok(ua) => ua,
+        Err(_) => unreachable!("The abstract path for the unix PID lock socket is static and should work - please report this as a bug"),
```

Comment:
I'ld rather not say unreachable here...this being true is up to kernel developers, not to us.

Reply:
I'm curious what you mean here. Are you referring to whether or not the kernel chooses to change the interface? This call in itself is preparing data for the kernel for sure, but it does not interact with the kernel at all. If you're making the argument that the kernel interface could change, I agree that it's possible. I'm open to making it an error if we want to fail without a panic if the kernel interface gets changed and nix decides to validate that too before passing it to the kernel, but it appears the only validation that this call does is on whether the path name is too long for the buffer. If you're concerned that nix might add additional validation, I'm happy to change this.

Reply:
I misread the message and thought you meant it was a property of the unix socket path, not of the string that _represents_ the unix socket path.

Using ```unreachable``` makes sense here. But since it really should be unreachable, please make the message more terse, something like: ```STRATISD_SOCKET_ADDR is a valid socket address name".

Reply:
Resolved.